### ISSUE TITLE
runtime: out of memory: cannot allocate 201326592-byte block (1306238976 in use)

### ISSUE THREAD

**Issue by mccoysc** [ethereum/go-ethereum#16824]
runtime: out of memory: cannot allocate 201326592-byte block (1306238976 in use)
fatal error: out of memory

runtime stack:
runtime.throw(0xcf85c0, 0xd)
        C:/Go/src/runtime/panic.go:616 +0x6b
runtime.largeAlloc(0xc000000, 0x35db0101, 0x450423)
        C:/Go/src/runtime/malloc.go:828 +0xdc
runtime.mallocgc.func1()
        C:/Go/src/runtime/malloc.go:721 +0x39
......

**Comment by mccoysc** [comment]
ver:1.8.9
windows 10 x64

**Comment by karalabe** [comment]
How much memory does your machine have? Are you sure you downloaded the 64bit build of Geth? What does `geth --version` print?

**Comment by tinshen** [comment]
you need more memory! 
I run geth server is 32g memory.

**Comment by stricq** [comment]
```
runtime: out of memory: cannot allocate 3221225472-byte block (31317721088 in use)
fatal error: out of memory

runtime stack:
runtime.throw(0xf140e2, 0xd)
        C:/go/src/runtime/panic.go:605 +0x9c
runtime.largeAlloc(0xc0000000, 0x5d70101, 0x459945)
        C:/go/src/runtime/malloc.go:829 +0x11b
runtime.mallocgc.func1()
        C:/go/src/runtime/malloc.go:722 +0x4d
runtime.systemstack(0xc04201f300)
        C:/go/src/runtime/asm_amd64.s:344 +0x7e
runtime.mstart()
        C:/go/src/runtime/proc.go:1135
```

I get this error running under Windows Server 2012 R2.  The machine has 128G of RAM with two CPUs.  geth has, essentially, access to pretty much all the RAM on the machine as very little else is running aside from the File and Storage Services.  This is the command line:

"C:\Users\Administrator\AppData\Roaming\Ethereum Wallet\binaries\Geth\unpacked\geth.exe" --syncmode fast --datadir C:\Users\Administrator\AppData\Roaming\Ethereum --cache 16384

**Comment by stricq** [comment]
It still happens under 1.8.10

```
runtime: out of memory: cannot allocate 3221225472-byte block (32171884544 in use)
fatal error: out of memory

runtime stack:
runtime.throw(0xeab47e, 0xd)
        C:/go/src/runtime/panic.go:616 +0x88
runtime.largeAlloc(0xc0000000, 0x450101, 0x2680000)
        C:/go/src/runtime/malloc.go:828 +0x117
runtime.mallocgc.func1()
        C:/go/src/runtime/malloc.go:721 +0x4d
runtime.systemstack(0x0)
        C:/go/src/runtime/asm_amd64.s:409 +0x7e
runtime.mstart()
        C:/go/src/runtime/proc.go:1175
```

**Comment by stricq** [comment]
This issue appears to be fixed with 1.8.11.  I've been running geth for about 24 hours now, it's memory usage has stayed under the 16G specified on the command line.  The block chain was about 500k behind (three weeks) and has just finished catching up.  No crashes at all, no high memory consumption.
